hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/metadata/HeadersPropagator.java

```diff
@@ -6,4 +6,8 @@
 public interface HeadersPropagator {
 
     Map<String, String> extract(Map<String, String> headers);
+
+    default Map<String, String> extractHeadersToLog(Map<String, String> headers) {
```

Comment:
I'm not sure if you need to add this method to all propagators. Not all propagators work in the context of logging/tracing. Maybe It would be better to introduce some new interface dedicated for logging/tracking. It is hard to me to distinguish which headers are for logging and which are propagated to the clients.